roles/StackStorm.st2smoketests/tasks/st2chatops.yml

```diff
@@ -1,15 +1,41 @@
 ---
+- name: Prepare temp file to track retry attempts
+  tempfile:
+    state: file
+  register: attempts_count
+  changed_when: no  # skip idempotence check
+
 - name: Verify st2chatops using bin/hubot
   # when editing, make sure it works for at least 2 adapters: 'shell' and 'slack'
-  shell: set -o pipefail && timeout 10 bash -c '(sleep 5; echo exit ) | bin/hubot'
+  shell: >
+    set -o pipefail &&
+    export n=0$(cat $attempts_count_file) &&
+    timeout $(( 9 + 6 ** n )) bash -c '(sleep $(( 4 + 6 ** n )); echo exit ) | bin/hubot';
+    echo $((n+1)) > $attempts_count_file
```

Comment:
would love to see some comments here on how this works, reading this it's pretty confusing and not very intuitive as to what it's doing.

Reply:
It was confusing to write too. Yeah, I'll add some comments.